From 485ce2c28cc3e1ba106df58534a1a13fa7e803b8 Mon Sep 17 00:00:00 2001
From: Szymon Kaliski <kaliskiszymon@gmail.com>
Date: Thu, 2 Mar 2017 12:33:52 +0100
Subject: [PATCH] borderless shadow

---
 sources/PseudoTerminal.m | 16 ++++++++++++++--
 1 file changed, 14 insertions(+), 2 deletions(-)

diff --git a/sources/PseudoTerminal.m b/sources/PseudoTerminal.m
index 84e9ada0..0e2613c4 100644
--- a/sources/PseudoTerminal.m
+++ b/sources/PseudoTerminal.m
@@ -386,7 +386,13 @@ static NSRect iTermRectCenteredVerticallyWithinRect(NSRect frameToCenter, NSRect
         case WINDOW_TYPE_LEFT_PARTIAL:
         case WINDOW_TYPE_RIGHT_PARTIAL:
         case WINDOW_TYPE_NO_TITLE_BAR:
-            return mask | NSBorderlessWindowMask | NSResizableWindowMask;
+            return (mask |
+                    NSTitledWindowMask |
+                    NSClosableWindowMask |
+                    NSMiniaturizableWindowMask |
+                    NSResizableWindowMask |
+                    NSTexturedBackgroundWindowMask |
+                    NSFullSizeContentViewWindowMask);
 
         case WINDOW_TYPE_TRADITIONAL_FULL_SCREEN:
             return mask | NSBorderlessWindowMask;
@@ -601,7 +607,7 @@ static NSRect iTermRectCenteredVerticallyWithinRect(NSRect frameToCenter, NSRect
         [myWindow setFrame:initialFrame display:NO];
     }
 
-    [myWindow setHasShadow:(windowType == WINDOW_TYPE_NORMAL)];
+    [myWindow setHasShadow:((windowType == WINDOW_TYPE_NORMAL) || (windowType == WINDOW_TYPE_NO_TITLE_BAR))];
 
     DLog(@"Create window %@", myWindow);
 
@@ -654,6 +660,12 @@ static NSRect iTermRectCenteredVerticallyWithinRect(NSRect frameToCenter, NSRect
     [self updateTabBarStyle];
     self.window.delegate = self;
 
+    if (windowType == WINDOW_TYPE_NO_TITLE_BAR) {
+        [[self window] setTitlebarAppearsTransparent:YES];
+        [[[self window] standardWindowButton:NSWindowCloseButton].superview.superview setHidden:YES];
+    }
+
+
     [[NSNotificationCenter defaultCenter] addObserver:self
                                              selector:@selector(updateWindowNumberVisibility:)
                                                  name:kUpdateLabelsNotification
-- 
2.12.0

